igvc_perception/src/multiclass_segmentation/README.md

```diff
@@ -0,0 +1,27 @@
+# Multiclass Semantic Segmentation System
+
+Using transfer learning, we can simultaneously classify lines and barrels in images. A U-Net with a pretrained EfficientNet encoder performs multiclass semantic segmentation, classifying the pixels in each image as a line, barrel, or neither. This system uses the [Catalyst](https://github.com/catalyst-team/catalyst) and [Segmentation_Models](https://github.com/qubvel/segmentation_models.pytorch) libraries. 
+
+<p align="center">
+  <img src="https://github.com/suhacker1/igvc-software/blob/multiclass_segmentation/igvc_perception/src/multiclass_segmentation/model_prediction.jpg">
+</p>
+
+## Folder Structure 
++ **IGVC_Full_Dataset.ipynb**: implements the training and testing of the neural network in a notebook
++ **IGVC_Full_Dataset.pdf**: provides a PDF version of the aforementioned notebook for readability
++ **data_loaders.py**: defines the Torch data loaders
++ **helper_operations.py**: defines multiple operations used for loss and metric calculations
++ **make_dataset.py**: converts folders of images and masks into .npy files
++ **requirements.txt**: lists dependencies
++ **segmentation_dataset.py**: defines the utilized dataset
++ **train.py**: implements, trains, and tests the model
+
+## Build Instructions 
+
+1. Download the dependencies in requirements.txt manually or using `pip install -r requirements.txt`. 
+2. Obtain a dataset containing the images (PNGs) and masks (JSONs). Skip steps #2 and #3 if using .npy files. 
+3. Run `python make_dataset.py -a /path_to_images(png) -b /path_to_masks(json)`to generate .npy files. Example: `python make_dataset.py -a /content/imgs/*.png -b /content/masks/*.json`
```

Comment:
I guess maybe you misunderstood what I meant. On cloud.robojackets.org, our dataset is currently stored like this:
```
Annotations
├── JESSII_COMPO_RUN_2_2019-06-08-16-38-22
├── JESSII_COMPO_RUN_3_2019-06-08-17-12-31
├── JESSII_DEBUG_1_2019-06-09-17-53-45
├── JESSII_DEBUG_2_2019-06-09-18-02-26
├── JESSII_FIRST_HEAT_RUN_ORIENTATION_BAD
├── JESSII_NIGHT_1_2019-06-09-22-56-22
├── JESSII_NIGHT_2_2019-06-09-23-04-39
├── JESSII_NIGHT_4_2019-06-09-23-15-06
├── JESSII_PRACTICE_FIELD_D_NO_PATH_2019-06-09-13-15-59
├── JESSII_PRACTICE_OLD_FOLLOWER_2019-06-09-16-16-30
├── JESSII_PRACTICE_U_CONE_2019-06-09-14-11-51
├── JESSII_PRACTICE_U_CONE_2_2019-06-09-14-26-13
├── JESSII_PRACTICE_U_CONE_3_2019-06-09-15-15-03
├── JESSII_PRACTICE_U_CONE_3_2019-06-09-15-16-16
├── JESSII_RAIN_LINES_2019-06-10-10-25-33
├── JESSII_RAMP_LINES_2019-06-09-18-43-55
├── JESSII_RUN_SUN_2_2019-06-10-15-18-43
└── JESSII_RUN_SUN_2_2019-06-10-15-30-13
```
where each one of those folders inside has like 
```
center_frame000088.json
center_frame000088.png
center_frame000104.json
center_frame000104.png
center_frame001050.json
center_frame001050.png
...
```

so, if you wanted to use your script with this type of folder structure, you would need to run it like
`python make_dataset.py -a annotations/**/*.png -b annotations/**/*.json`

Reply:
Will fix

Reply:
Just committed a potential fix